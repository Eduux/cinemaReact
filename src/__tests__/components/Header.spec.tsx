import React from 'react';
import { render } from '@testing-library/react';

import Component from 'components/Header';

import withMovies from 'containers/movies';

const Container = withMovies(Component);

jest.mock('react-router-dom', () => ({
  useHistory: jest.fn(),
  NavLink: ({ children }: { children: React.ReactNode }) => children,
}));

describe('Header component', () => {
  it('should render', () => {
    const result = render(<Container />);
    expect(result).toMatchSnapshot();
  });
});
