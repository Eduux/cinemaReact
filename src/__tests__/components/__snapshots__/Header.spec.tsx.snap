// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-jRQAMF eXNhWB"
      >
        <div
          class="sc-gKckTs cAsVgv"
        >
          Home
          <div
            class="sc-gsDJrp fVOCov"
          >
            <form
              class="sc-hKwCoD cMFRUd"
            >
              <input
                class="sc-dkPtyc ekUMin"
                name="search"
                placeholder="Procurar filme:"
                value=""
              />
              <div
                class="sc-eCImvq NPLqQ"
              >
                <button
                  class="sc-bdvvaa gWolTY"
                  type="submit"
                  width="90px"
                >
                  Buscar
                </button>
              </div>
            </form>
          </div>
          Favoritos
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-jRQAMF eXNhWB"
    >
      <div
        class="sc-gKckTs cAsVgv"
      >
        Home
        <div
          class="sc-gsDJrp fVOCov"
        >
          <form
            class="sc-hKwCoD cMFRUd"
          >
            <input
              class="sc-dkPtyc ekUMin"
              name="search"
              placeholder="Procurar filme:"
              value=""
            />
            <div
              class="sc-eCImvq NPLqQ"
            >
              <button
                class="sc-bdvvaa gWolTY"
                type="submit"
                width="90px"
              >
                Buscar
              </button>
            </div>
          </form>
        </div>
        Favoritos
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
